test-data/unit/check-typeddict.test

```diff
@@ -610,3 +610,25 @@ def f() -> None:
     A = TypedDict('A', {'x': int})
 A  # E: Name 'A' is not defined
 [builtins fixtures/dict.pyi]
+
+
+-- Use dict literals
+
+[case testTypedDictDictLiterals]
+from mypy_extensions import TypedDict
+
+Point = TypedDict('Point', {'x': int, 'y': int})
+
+def f(p: Point) -> None:
+    p = {'x': 2, 'y': 3}
+    p = {'x': 2}  # E: Expected items ['x', 'y'] but found ['x'].
+    p = dict(x=2, y=3)
+
+f({'x': 1, 'y': 3})
+f({'x': 1, 'y': 'z'})  # E: Incompatible types (expression has type "str", TypedDict item "y" has type "int")
+
+f(dict(x=1, y=3))
+f(dict(x=1, y=3, z=4))  # E: Expected items ['x', 'y'] but found ['x', 'y', 'z'].
```

Comment:
I would add one more test case with few tests where you assign to variables with explicit types, like:
```python
p: Point = {'x', 'hi'}
```
and/or
```python
p: Point
p = dict(x='bye')
```